src/sagemaker/hyperpod/inference/hp_endpoint_base.py

```diff
@@ -114,7 +114,10 @@ def call_get_api(
                 name=name,
             )
         except Exception as e:
-            handle_exception(e, name, namespace)
+            # Map kind to correct resource type
+            resource_type = 'hyp_jumpstart_endpoint' if kind == 'JumpStartModel' else 'hyp_custom_endpoint'
```

Comment:
this stems from a lack of a single entrypoin, your decorator is able to deduce this